pkg/registry/core/pod/strategy.go

```diff
@@ -89,6 +89,8 @@ func (podStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 	podutil.DropDisabledPodFields(pod, nil)
 
 	applyWaitingForSchedulingGatesCondition(pod)
+	mutatePodAffinity(pod)
+	mutatePodTopologySpread(pod)
```

Comment:
PodTopologySpread's matchLabels field should work without this change, right? I guess you're trying to unify the internal mechanics for (mis)matchLabelKeys, but it's better to only include changes related with podAffinity in this PR.

Reply:
> it's better to only include changes related with podAffinity in this PR.

OK, let me revert changes for the TopologySpread.

Reply:
thanks, it can be tracked in a follow-up.